brave/browser/net/proxy/proxy_config_service_tor.cc

```diff
@@ -4,32 +4,142 @@
 
 #include "brave/browser/net/proxy/proxy_config_service_tor.h"
 
+#include <memory>
+#include <utility>
 #include <vector>
 
+#include "base/path_service.h"
 #include "base/strings/string_number_conversions.h"
 #include "base/strings/string_util.h"
+#include "chrome/common/chrome_paths.h"
 #include "crypto/random.h"
-#include "url/gurl.h"
+#include "net/proxy/proxy_service.h"
+#include "net/url_request/url_request_context.h"
+#include "services/service_manager/public/cpp/connector.h"
 
 namespace net {
 
+using content::BrowserThread;
 const int kTorPasswordLength = 16;
 
-ProxyConfigServiceTor::ProxyConfigServiceTor(const GURL& url)
-  : url_(url) {
-    config_.proxy_rules().ParseFromString(url_.spec());
+ProxyConfigServiceTor::ProxyConfigServiceTor(const std::string tor_path,
```

Comment:
sorry, I missed this. Next time remember to use passed by reference to avoid entire object copy